File: Models/Knjiga.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace Models
{
    public enum TipKnjige
    {   ZaDecu,
        Lektira,
        Price,
        Poezija,
        Roman
    }
    [Table("Knjige")]
    public class Knjiga
    {
        [Key]
        public int IdKnjige { get; set; }
        
        [MaxLength(50)]
        [Required]
        public string Naslov { get; set; } 

        [Required]
        public int BrojPrimeraka { get; set; }

        [Required]
        public int BrojStrana { get; set; }

        [Required]
        public TipKnjige Zanr { get; set; }

        [JsonIgnore]
        public List<Iznajmljivanje> Izdate { get; set; }

        //[JsonIgnore]
        public Autor Autor { get; set; }

    }
}

File: Klijent/Knjiga.js
export class Knjiga
{
    constructor(naslov, autor, zanr, brp, brs)
    {
        this.naslov=naslov;
        this.autor=autor;
        this.zanr=zanr;
        this.brojPrimeraka=brp;
        this.brojStrana=brs;
    }

    crtaj(host) {

        var tr = document.createElement("tr");
        host.appendChild(tr);

        var el;
        el = document.createElement("td");
        el.innerHTML = this.naslov;
        tr.appendChild(el);

        el = document.createElement("td");
        el.innerHTML = this.autor.ime +" "+ this.autor.prezime;
        tr.appendChild(el);

        el = document.createElement("td");
        var zanrovi=["Za decu","Lektira","Price","Poezija","Roman"];
        el.innerHTML = zanrovi[this.zanr];
        tr.appendChild(el);

        el = document.createElement("td");
        el.innerHTML = this.brojPrimeraka;
        tr.appendChild(el);

        el = document.createElement("td");
        el.innerHTML = this.brojStrana;
        tr.appendChild(el);
      
    }

}